cpp-client/deephaven/client/include/private/deephaven/client/subscription/subscription_handle.h

```diff
@@ -7,6 +7,11 @@ namespace deephaven::client::subscription {
 class SubscriptionHandle {
 public:
   virtual ~SubscriptionHandle() = default;
-  virtual void cancel() = 0;
+  /**
+   * Cancels the subscription.
+   * @param wait If true, waits for the internal subcription thread to be torn down. Use 'true'
+   * if you want to be sure that your callback will not be invoked after this call returns.
+   */
+  virtual void cancel(bool wait) = 0;
```

Comment:
Would it make sense to have two calls, `cancel_sync` and `cancel_async` ?
Lately I've been leaning on being more explicit in naming as opposed to parameters that users may copy/paste/miss...

Reply:
Let's discuss offline and decide. I have some opinions.